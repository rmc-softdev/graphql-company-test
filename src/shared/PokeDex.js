import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import PropTypes from "prop-types";

const PokeDex = ({ closeModalHandler }) => {
  const caughtPokemons = useSelector((state) => state.caughtPokemons);
  const dispatch = useDispatch();

  const deleteHandler = (poke) => {
    dispatch({
      type: "DELETE_POKE",
      payload: poke,
    });
  };

  return (
    <>
      <div
        style={{
          textAlign: "center",
          margin: "10px 0",
          fontWeight: 700,
          marginTop: "30px",
        }}
      >
        {" "}
        My PokeDex
      </div>
      <div
        className="results showcase"
        style={{ margin: "0 auto", boxShadow: "none" }}
      >
        {caughtPokemons?.length !== 0 ? (
          <ul style={{ padding: 0 }}>
            {caughtPokemons?.map((poke) => (
              <li key={poke?.id}>
                <div
                  className="results-items"
                  style={{
                    padding: "3px 0",
                    borderBottom: "1px solid #cacaca",
                  }}
                >
                  <div className="result content">
                    <div style={{ position: "relative" }}>
                      <span
                        className="pokemon__delete-btn"
                        onClick={() => deleteHandler(poke)}
                      >
                        {" "}
                        x{" "}
                      </span>
                      <Link
                        to={`/pokemon/${poke?.name?.toLowerCase()}`}
                        onClick={closeModalHandler}
                      >
                        {" "}
                        <img src={poke?.image} alt="" />{" "}
                      </Link>
                    </div>
                    <span className="resultsName">
                      <Link
                        to={`/pokemon/${poke?.name?.toLowerCase()}`}
                        onClick={closeModalHandler}
                      >
                        {" "}
                        {poke?.name}{" "}
                      </Link>
                    </span>
                  </div>
                  <div className="results media">{poke?.number}</div>
                </div>
              </li>
            ))}
          </ul>
        ) : (
          <div style={{ padding: "10px", margin: "10px 0" }}>
            {" "}
            You have no pokemons. Maybe catch one?
          </div>
        )}
      </div>
    </>
  );
};

PokeDex.propTypes = {
  closeModalHandler: PropTypes.func.isRequired,
};

export default PokeDex;
