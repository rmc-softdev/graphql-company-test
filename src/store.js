import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";

const initialState = {
  caughtPokemons: [],
};

const middleware = [thunk];

if (localStorage.getItem("caughtpokes") === null) {
  localStorage.setItem("caughtpokes", JSON.stringify(initialState));
} else {
  initialState.caughtPokemons = JSON.parse(localStorage.getItem("caughtpokes"));
}

const rootReducer = (state = initialState, action) => {
  let stateCopy;
  switch (action.type) {
    case "ADD_POKE":
      stateCopy = [...state.caughtPokemons, action.payload];
      localStorage.setItem("caughtpokes", JSON.stringify(stateCopy));
      return {
        ...state,
        caughtPokemons: [...state.caughtPokemons, action.payload],
      };
    case "DELETE_POKE":
      stateCopy = [
        ...state.caughtPokemons.filter((el) => {
          return el.name !== action.payload.name;
        }),
      ];
      localStorage.setItem("caughtpokes", JSON.stringify(stateCopy));
      return {
        ...state,
        caughtPokemons: [
          ...state.caughtPokemons.filter((el) => {
            return el.name !== action.payload.name;
          }),
        ],
      };
    default:
      return state;
  }
};

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
