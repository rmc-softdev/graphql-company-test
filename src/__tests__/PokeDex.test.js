import React from "react";
import { render } from "@testing-library/react";
import PokeDex from "../shared/PokeDex";
import { Provider } from "react-redux";
import store from "../store";

const mockPokemons = [
  {
    id: "UG9rZW1vbjowMDE=",
    name: "Bulbasaur",
    image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
    number: "001",
    types: ["Grass", "Poison"],
    __typename: "Pokemon",
  },
];

it("should renders an array of pokemons objects", () => {
  // const mockAddPoke = jest.fn()
  // const mockDeletePoke = jest.fn()
  const RenderedPokes = render(
    <Provider store={store}>
      <PokeDex />
    </Provider>
  );
  expect(RenderedPokes).toMatchSnapshot();
});
