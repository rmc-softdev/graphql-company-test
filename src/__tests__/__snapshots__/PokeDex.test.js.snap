// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders an array of pokemons objects 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="text-align: center; margin: 30px 0px 10px 0px; font-weight: 700;"
      >
         
        My PokeDex
      </div>
      <div
        class="results showcase"
        style="margin: 0px auto; box-shadow: none;"
      >
        <div
          style="padding: 10px; margin: 10px 0px;"
        >
           
          You have no pokemons. Maybe catch one?
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="text-align: center; margin: 30px 0px 10px 0px; font-weight: 700;"
    >
       
      My PokeDex
    </div>
    <div
      class="results showcase"
      style="margin: 0px auto; box-shadow: none;"
    >
      <div
        style="padding: 10px; margin: 10px 0px;"
      >
         
        You have no pokemons. Maybe catch one?
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should renders an array of todo objects 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="text-align: center; margin: 30px 0px 10px 0px; font-weight: 700;"
      >
         
        My PokeDex
      </div>
      <div
        class="results showcase"
        style="margin: 0px auto; box-shadow: none;"
      >
        <div
          style="padding: 10px; margin: 10px 0px;"
        >
           
          You have no pokemons. Maybe catch one?
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="text-align: center; margin: 30px 0px 10px 0px; font-weight: 700;"
    >
       
      My PokeDex
    </div>
    <div
      class="results showcase"
      style="margin: 0px auto; box-shadow: none;"
    >
      <div
        style="padding: 10px; margin: 10px 0px;"
      >
         
        You have no pokemons. Maybe catch one?
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
